File: Client_Root/Client/Assets/Scripts/Network/Messages/ToRoom/ReadyForStartToR.cs
using FlatBuffers;

public class ReadyForStartToR : IMessage
{
    public const ushort MESSAGE_ID = MessageID.ReadyForStartToR_ID;

    public int m_nPlayerIndex;

    public override ushort GetID()
    {
        return MESSAGE_ID;
    }

    public override IMessage Clone()
    {
        return null; 
    }

    public override byte[] Serialize()
    {
        FBS.ReadyForStartToR.StartReadyForStartToR(m_Builder);
        FBS.ReadyForStartToR.AddPlayerIndex(m_Builder, m_nPlayerIndex);
        var data = FBS.ReadyForStartToR.EndReadyForStartToR(m_Builder);

        m_Builder.Finish(data.Value);

        return m_Builder.SizedByteArray();
    }

    public override bool Deserialize(byte[] bytes)
    {
        var buf = new ByteBuffer(bytes);

        var data = FBS.ReadyForStartToR.GetRootAsReadyForStartToR(buf);

        m_nPlayerIndex = data.PlayerIndex;

        return true;
    }
}

File: Server_Root/Server/RoomServer/Messages/ToRoom/ReadyForStartToR.cpp
#include "stdafx.h"
#include "ReadyForStartToR.h"
#include "../../../FBSFiles/ReadyForStartToR_generated.h"

ReadyForStartToR::ReadyForStartToR()
{
}

ReadyForStartToR::~ReadyForStartToR()
{
}

unsigned short ReadyForStartToR::GetID()
{
	return MESSAGE_ID;
}

IMessage* ReadyForStartToR::Clone()
{
	return NULL;
}

const char* ReadyForStartToR::Serialize(int* pLength)
{
	FBS::ReadyForStartToRBuilder data_builder(m_Builder);
	data_builder.add_PlayerIndex(m_nPlayerIndex);
	auto data = data_builder.Finish();

	m_Builder.Finish(data);

	*pLength = m_Builder.GetSize();

	return (char*)m_Builder.GetBufferPointer();
}

bool ReadyForStartToR::Deserialize(const char* pChar)
{
	auto data = flatbuffers::GetRoot<FBS::ReadyForStartToR>((const void*)pChar);

	m_nPlayerIndex = data->PlayerIndex();

	return true;
}